### PR TITLE
[AKS] `az aks create/update`: Add new parameters `--enable-gmsa`, `--gmsa-dns-server`, `--gmsa-root-domain-name` to support Windows gMSA v2

### PR DESCRIPTION
**Pull request by FumingZhang**
**Description**

`az aks create/update`: Add new parameters `--enable-gmsa`, `--gmsa-dns-server`, `--gmsa-root-domain-name` to support Windows gMSA v2.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change.
[Component Name 2] `az command b`: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py [hunk 1]
```diff
@@ -4702,16 +4868,28 @@ def set_up_windows_profile(self, mc: ManagedCluster) -> ManagedCluster:
             windows_admin_password,
         ) = self.context.get_windows_admin_username_and_password()
         if windows_admin_username or windows_admin_password:
+            # license
             windows_license_type = None
             if self.context.get_enable_ahub():
                 windows_license_type = "Windows_Server"
 
+            # gmsa
+            gmsa_profile = None
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/decorator.py:4877]: User is allowed to set the configuration of gmsa only after windows admin user name/password is provided.

If neither user name nor password is provided, `--enable-gmsa/--gmsa-dns-server/--gmsa-root-domain-name` would be ignored (same behavior as in aks-preview).

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py [hunk 2]
```diff
@@ -5811,8 +5990,9 @@ def update_windows_profile(self, mc: ManagedCluster) -> ManagedCluster:
         enable_ahub = self.context.get_enable_ahub()
         disable_ahub = self.context.get_disable_ahub()
         windows_admin_password = self.context.get_windows_admin_password()
+        enable_windows_gmsa = self.context.get_enable_windows_gmsa()
 
-        if any([enable_ahub, disable_ahub, windows_admin_password]) and not mc.windows_profile:
+        if any([enable_ahub, disable_ahub, windows_admin_password, enable_windows_gmsa]) and not mc.windows_profile:
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/decorator.py:5995]: Users can only change the configuration of gmsa when they update the cluster if they set the windows profile when the cluster is created.

If the user does not set the windows-related configuration when creating the cluster, the configuration of gmsa cannot be changed through update.

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py (hunk 1 above)
Comment (AbelHu) [reply src/azure-cli/azure/cli/command_modules/acs/decorator.py:4877]: I think that we should raise an error instead of ignoring it. how do you think?

**Review by AbelHu — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py (hunk 1 above)
Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/decorator.py:4894]: OK, updated in commit [#dc72c59](https://github.com/Azure/azure-cli/pull/21472/commits/dc72c5971946cd0d9d3ac78bc846a0c14cdda14a).

**Review by andyliuliming — APPROVED** [review]

**Pull request merged**